src/compartment.c

```diff
@@ -981,47 +1030,74 @@ extract_sym_offset(struct Compartment *comp, struct LibSymSearchResult res)
 
 static struct LibSymSearchResult
 find_lib_dep_sym_in_comp(const char *to_find,
-    struct Compartment *comp_to_search, const unsigned short sym_type)
+    struct Compartment *comp_to_search, const size_t source_lib_idx,
+    const unsigned short sym_type, bool in_lib)
 {
-    for (size_t i = 0; i < comp_to_search->libs_count; ++i)
+    struct LibSymSearchResult res = { -1, -1, false };
+    if (in_lib)
+    {
+        res = find_lib_dep_sym_in_lib(
+            to_find, comp_to_search, source_lib_idx, sym_type);
+    }
+    else
     {
-        for (size_t j = 0; j < comp_to_search->libs[i]->lib_syms_count; ++j)
+        for (size_t i = 0; i < comp_to_search->libs_count; ++i)
         {
-            // Ignore non-symbol relocations
-            if (!comp_to_search->libs[i]->lib_syms[j].sym_name)
+            if (i == source_lib_idx)
             {
                 continue;
             }
+            res = find_lib_dep_sym_in_lib(to_find, comp_to_search, i, sym_type);
+            if (res.found)
+            {
+                break;
+            };
```

Comment:
Superfluous `;` I think.

Reply:
Deleted in https://github.com/capablevms/CHERI-ELF-comp/pull/33/commits/1482d4175f08eec6c7ade8e61405266f79ef3bfa.